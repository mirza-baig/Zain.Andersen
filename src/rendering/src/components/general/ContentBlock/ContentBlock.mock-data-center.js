const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'ContentBlock',
  },
  fields: {
    componentSpacing: {
      id: '983f6ac4-9baa-4ddf-9272-349f50760fd2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Component-Spacing/Standard',
      name: 'Standard',
      displayName: 'Standard',
      fields: {
        Value: {
          value: 'standard',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    sectionId: {
      value: 'content block left',
    },
    theme: {
      value: '',
    },
    backgroundColor: {
      id: '6c8aa36e-f22d-400a-9fe7-391181db23ad',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Background-Color/Black',
      name: 'Black',
      displayName: 'Black',
      fields: {
        Value: {
          value: 'black',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    alignment: {
      id: 'ed4b79ff-078b-47e0-8374-d30cd00224c9',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Alignment/Center',
      name: 'Center',
      displayName: 'Center',
      fields: {
        Value: {
          value: 'center',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value: 'This is a headline lorem ipsum dolor amit',
    },
    body: {
      value:
        'Purus sit amet volutpat consequat mauris nunc congue nisi. Eget arcu dictum varius duis at consectetur. Mi proin sed libero enim sed faucibus turpis in. Feugiat vivamus at augue eget arcu dictum varius. Urna cursus eget nunc scelerisque viverra mauris in aliquam. Sem integer vitae justo eget magna fermentum.\n<br />\n<br />\nLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Enim sed faucibus turpis in eu mi bibendum neque egestas. Pharetra diam sit amet nisl suscipit adipiscing. Mi tempus imperdiet nulla malesuada pellentesque. Duis at consectetur lorem donec massa sapien faucibus et molestie. Risus quis varius quam quisque id diam vel quam elementum.',
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Link: {
      value: {
        href: '/',
        text: 'Call to Action',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta1Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'primary',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Link: {
      value: {
        href: '/',
        text: 'Call to Action',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta2Style: {
      id: 'dd818850-ec95-4d32-9774-7cc8173e277b',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Link',
      name: 'Link',
      displayName: 'Link',
      fields: {
        Value: {
          value: 'link',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
