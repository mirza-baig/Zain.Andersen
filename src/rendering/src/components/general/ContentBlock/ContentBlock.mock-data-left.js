const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'ContentBlock',
  },
  fields: {
    componentSpacing: {
      id: '983f6ac4-9baa-4ddf-9272-349f50760fd2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Component-Spacing/Standard',
      name: 'Standard',
      displayName: 'Standard',
      fields: {
        Value: {
          value: 'standard',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    sectionId: {
      value: 'content block left',
    },
    theme: {
      value: '',
    },
    backgroundColor: {
      id: '64333070-0d1b-4a33-9e19-7d299d6e8dd3',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Background-Color/White',
      name: 'White',
      displayName: 'White',
      fields: {
        Value: {
          value: 'white',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    alignment: {
      id: 'c6dc5505-a8a3-47e0-8b10-f01b2c03aa32',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Alignment/Left',
      name: 'Left',
      displayName: 'Left',
      fields: {
        Value: {
          value: 'left',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value: 'This is a headline lorem ipsum dolor amit',
    },
    body: {
      value:
        'Purus sit amet volutpat consequat mauris nunc congue nisi. Eget arcu dictum varius duis at consectetur. Mi proin sed libero enim sed faucibus turpis in. Feugiat vivamus at augue eget arcu dictum varius. Urna cursus eget nunc scelerisque viverra mauris in aliquam. Sem integer vitae justo eget magna fermentum.\n<br />\n<br />\nLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Enim sed faucibus turpis in eu mi bibendum neque egestas. Pharetra diam sit amet nisl suscipit adipiscing. Mi tempus imperdiet nulla malesuada pellentesque. Duis at consectetur lorem donec massa sapien faucibus et molestie. Risus quis varius quam quisque id diam vel quam elementum.',
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Link: {
      value: {
        href: '/',
        text: 'Call to Action',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta1Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'primary',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Link: {
      value: {
        href: '/',
        text: 'Call to Action',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta2Style: {
      id: 'dd818850-ec95-4d32-9774-7cc8173e277b',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Link',
      name: 'Link',
      displayName: 'Link',
      fields: {
        Value: {
          value: 'link',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
