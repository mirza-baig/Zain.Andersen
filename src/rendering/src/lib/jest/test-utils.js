// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
