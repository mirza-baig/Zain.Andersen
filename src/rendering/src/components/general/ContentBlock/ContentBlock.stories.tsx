// Global
import { Story, Meta } from '@storybook/react';

// Local
import ContentBlock, { ContentBlockProps } from './ContentBlock';
import defaultDataCenter from './ContentBlock.mock-data-center';
import defaultDataLeft from './ContentBlock.mock-data-left';
import defaultDataGray from './ContentBlock.mock-data-gray';

export default {
  title: 'General/ContentBlock',
  component: ContentBlock,
} as Meta;

const Template: Story<ContentBlockProps> = (props) => <ContentBlock {...props} />;

export const DefaultCenter = Template.bind({});
DefaultCenter.args = defaultDataCenter;

export const DefaultLeft = Template.bind({});
DefaultLeft.args = defaultDataLeft;

export const DefaultGray = Template.bind({});
DefaultGray.args = defaultDataGray;
