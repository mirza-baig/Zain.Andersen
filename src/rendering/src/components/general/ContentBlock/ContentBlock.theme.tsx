import classNames from 'classnames';
import { ThemeFile } from 'lib/context/ThemeContext';
import { BackgroundColor } from './ContentBlock';
import { ButtonVariants } from 'src/helpers/Button';

const getDynamicButtonStyles = (backgroundColor: BackgroundColor): string => {
  return classNames(`${backgroundColor === 'black' ? 'text-primary' : ''}`);
};

const getSecondCTAPadding = (ctaStyle: ButtonVariants) => {
  return (ctaStyle === 'link' || ctaStyle === 'link-right-icon') && 'ml-xs md:ml-0 px-s md:px-0';
};

export const ContentBlockTheme = (
  backgroundColor: BackgroundColor,
  cta2Style: ButtonVariants
): ThemeFile => {
  return {
    aw: {
      classes: {
        contentWrapper: classNames(
          backgroundColor != 'white' ? 'py-l' : '',
          'col-span-12',
          'px-m',
          'md:w-full',
          'md:max-w-screen-lg',
          'md:mx-auto'
        ),
        headlineClass: 'text-theme-text text-sm-m md:text-m font-bold mb-s',
        bodyClass: 'text-theme-body mb-s',
        buttonGroupClass: {
          wrapper: 'flex-col',
          cta1Classes: classNames(getDynamicButtonStyles(backgroundColor), 'mb-m md:mb-0'),
          cta2Classes: classNames(
            getDynamicButtonStyles(backgroundColor),
            getSecondCTAPadding(cta2Style)
          ),
          cta3Classes: ' mb-m md:mb-0',
        },
      },
    },
    rba: {
      classes: {
        contentWrapper: classNames(
          backgroundColor != 'white' ? 'py-l' : '',
          'col-span-12',
          'px-m',
          'md:w-full',
          'md:max-w-screen-lg',
          'md:mx-auto'
        ),
        headlineClass: 'text-theme-text text-sm-m md:text-m font-medium mb-s',
        bodyClass: 'text-theme-body mb-s',
        buttonGroupClass: {
          wrapper: 'flex-col md:items-center',
          cta1Classes: classNames(getDynamicButtonStyles(backgroundColor), 'mr-2'),
          cta2Classes: classNames(getDynamicButtonStyles(backgroundColor), 'my-s md:my-0'),
        },
      },
    },
  };
};
