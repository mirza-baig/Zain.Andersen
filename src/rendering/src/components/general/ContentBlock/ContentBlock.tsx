// Components
import { BodyCopy } from 'src/helpers/BodyCopy';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { Component } from 'src/helpers/Component';
import { ContentBlockTheme } from './ContentBlock.theme';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { getEnum } from 'lib/utils';
import { Headline } from 'src/helpers/Headline';
import { useTheme } from 'lib/context/ThemeContext';
import { Button, ButtonVariants } from 'src/helpers/Button';
import classNames from 'classnames';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';

export type ContentBlockProps =
  Feature.EnterpriseWeb.Enterprise.Components.General.ContentBlock.ContentBlock;

export type BackgroundColor = 'black' | 'gray' | 'white';

const ContentBlock = (props: ContentBlockProps): JSX.Element => {
  const backgroundColor = getEnum<BackgroundColor>(props.fields?.backgroundColor) || 'white';
  const legalCopyFont = props?.fields?.useLegalCopyFont?.value || false;
  const { themeData } = useTheme(
    ContentBlockTheme(backgroundColor, getEnum<ButtonVariants>(props.fields?.cta2Style) || 'link')
  );

  return (
    <Component
      variant="full"
      gap="gap-x-0"
      padding="px-0"
      backgroundVariant={getEnum(props.fields?.backgroundColor)}
      dataComponent="general/contentblock"
      {...props}
    >
      <div className={themeData.classes.contentWrapper}>
        <Headline defaultTag="h2" classes={themeData.classes.headlineClass} {...props} />
        <BodyCopy
          classes={classNames(themeData.classes.bodyClass)}
          refer={legalCopyFont ? 'legal-copy' : 'body-copy'}
          {...props}
        />
        <div
          className={
            props.fields.ctaAlignment?.displayName == 'Stack'
              ? 'flex flex-col items-start  md:space-y-4'
              : 'flex flex-col items-start  md:flex-row md:space-x-10'
          }
        >
          <ButtonGroup
            classes={{
              wrapper: themeData.classes.buttonGroupClass.wrapper,
              cta1Classes: themeData.classes.buttonGroupClass.cta1Classes,
              cta2Classes: themeData.classes.buttonGroupClass.cta2Classes,
              cta3Classes: themeData.classes.buttonGroupClass.cta3Classes,
              ctaAlignment: props.fields.ctaAlignment?.displayName == 'Stack',
            }}
            {...props}
          />


        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()(ContentBlock);
