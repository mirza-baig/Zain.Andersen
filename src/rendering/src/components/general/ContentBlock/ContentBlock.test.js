// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import ContentBlock from './ContentBlock';
import defaultData from './ContentBlock.mock-data-center';

it('renders correctly', async () => {
  let component = snapshot(ContentBlock, { componentProps: defaultData });
  await hasDataComponent(component, 'general/contentblock');
});
