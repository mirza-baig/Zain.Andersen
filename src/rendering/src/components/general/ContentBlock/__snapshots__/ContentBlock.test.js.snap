// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/contentblock"
  id="content block left"
>
  <div
    class="my-8 w-full theme-black bg-theme-bg"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-x-0 "
    >
      <div
        class="py-l col-span-12 px-m md:w-full md:max-w-screen-lg md:mx-auto"
      >
        <div
          class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-m md:text-m font-bold mb-s"
        >
          <h2>
            This is a headline lorem ipsum dolor amit
          </h2>
        </div>
        <div
          class="text-theme-body mb-s"
        >
          <div
            class="body-copy"
          >
            <div>
              Purus sit amet volutpat consequat mauris nunc congue nisi. Eget arcu dictum varius duis at consectetur. Mi proin sed libero enim sed faucibus turpis in. Feugiat vivamus at augue eget arcu dictum varius. Urna cursus eget nunc scelerisque viverra mauris in aliquam. Sem integer vitae justo eget magna fermentum.

              <br />
              

              <br />
              
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Enim sed faucibus turpis in eu mi bibendum neque egestas. Pharetra diam sit amet nisl suscipit adipiscing. Mi tempus imperdiet nulla malesuada pellentesque. Duis at consectetur lorem donec massa sapien faucibus et molestie. Risus quis varius quam quisque id diam vel quam elementum.
            </div>
          </div>
        </div>
        <div
          class="flex flex-col items-start  md:flex-row md:space-x-10"
        >
          <div
            class="flex-col mb-s flex items-start md:flex-row md:items-center md:space-x-4"
          >
            <a
              aria-label="Call to Action"
              class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed text-primary mb-m md:mb-0 mr-m"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/"
              role="link"
              tabindex="0"
              title=""
            >
              Call to Action
              <span
                class="ml-xxs"
              />
            </a>
            <a
              aria-label="Call to Action"
              class="hover:decoration-theme-btn-decoration flex w-fit items-center whitespace-normal font-sans text-text-link font-heavy text-theme-text hover:underline hover:underline-offset-8 disabled:border-gray disabled:text-gray text-primary ml-xs md:ml-0 px-s md:px-0"
              data-gtm-click=""
              data-gtm-dl-event="cta_click"
              href="/"
              role="link"
              tabindex="0"
              title=""
            >
              Call to Action
              <span
                class="ml-xxs"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
